arch/arm/src/tlsr82/tlsr82_serial.c

```diff
@@ -612,14 +612,20 @@ static inline uint8_t uart_read_byte(int uart_num)
 
 static inline void uart_send_byte(uint8_t byte)
 {
+  irqstate_t flags;
+
   while (UART_GET_TX_BUF_CNT() > 7);
 
+  flags = enter_critical_section();
+
   UART_BUF(uart_txindex) = byte;
```

Comment:
is it reasonable to check `UART_GET_TX_BUF_CNT() <= 7` after entering critical section and jump to `while (UART_GET_TX_BUF_CNT() > 7);` in case if it is false? I mean trying to handle the case when preemption happens right before `flags = enter_critical_section();` and other writer puts a char.

Reply:
Ack and Done.